### PR TITLE
[feature](tvf) support query table value function

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

This PR supports a Table Value Function called `Query`. He can push a query directly to the catalog source for execution by specifying `catalog` and `query` without parsing by Doris. Doris only receives the results returned by the query.
Currently only JDBC Catalog is supported.

Example:

```
Doris > desc function query('catalog' = 'mysql','query' = 'select count(*) as cnt from test.test');           
+-------+--------+------+------+---------+-------+
| Field | Type   | Null | Key  | Default | Extra |
+-------+--------+------+------+---------+-------+
| cnt   | BIGINT | Yes  | true | NULL    | NONE  |
+-------+--------+------+------+---------+-------+

Doris > select * from query('catalog' = 'mysql','query' = 'select count(*) as cnt from test.test');           
+----------+
| cnt      |
+----------+
| 30000000 |
+----------+
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/QueryTableValueFunction.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            throws AnalysisException {
+        String catalogName = params.get(CATALOG);
+        CatalogIf catalogIf = Env.getCurrentEnv().getCatalogMgr().getCatalog(catalogName);
+        if (catalogIf == null) {
+            throw new AnalysisException("Catalog not found: " + catalogName);
+        }
+        if (catalogIf instanceof JdbcExternalCatalog) {
+            return new JdbcQueryTableValueFunction(params);
+        } else {
+            throw new AnalysisException("Unsupported catalog type in query tvf: " + catalogName);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/QueryTableValueFunction.java:65]: print "type" as well

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/QueryTableValueFunction.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        }
+        if (!params.containsKey(CATALOG) || !params.containsKey(QUERY)) {
+            throw new AnalysisException("Query TableValueFunction must have 2 arguments: 'catalog' and 'query'");
+        }
+        String catalogName = params.get(CATALOG);
+        this.query = params.get(QUERY);
+        this.catalogIf = Env.getCurrentEnv().getCatalogMgr().getCatalog(catalogName);
+    }
+
+    public static QueryTableValueFunction createQueryTableValueFunction(Map<String, String> params)
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/QueryTableValueFunction.java:55]: Missing authorization check.
I think user need catalog level SELECT priv to exeute this tvf

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Pull request merged**